Create rule S6487: "std::format" should not be missing indexes (CPP-3029) (#1448)
SonarSource/rspec: rules/S6487/cfamily/rule.adoc

> When using `std::format`, you pass to it a format string that contains replacement fields (surrounded with `{}`)

Reviewer: ```suggestion
`std::format` takes as an argument a format string that contains replacement fields (surrounded with `{}`)
```

> and some extra arguments that will be formatted inside these replacement fields. 

Reviewer: ```suggestion
and a set of extra arguments that will be formatted inside the replacement fields. 
```

Even if the format string is checked at compile-time, it is possible to have a mismatch between the format string and the arguments. For example, when:

* The format string contains fewer replacement fields than the number of extra arguments:
  `std::format("{} {}", 1, 2, 3);`
* The format string uses indexes for the replacement fields, but one index is missing:
  `std::format("{0} {0} {2}", 1, 2, 3);`

> In these cases, the extra arguments are silently ignored. In the best case, it is dead code that should be removed.

Reviewer: ```suggestion
In these cases, the extra arguments are silently ignored. In the best-case scenario, it leads to dead code.
```

> Otherwise, it is a typo, and the output will not be the expected one.

Reviewer: ```suggestion
Otherwise, it is a typo, and the output will not be intended.
```

> When using `std::format`, you pass it a format string that contains replacement fields (surrounded with `{}`)

Reviewer: ```suggestion
When using `std::format`, you pass to it a format string that contains replacement fields (surrounded with `{}`)
```

> and some extra arguments that will be formatted inside those replacement fields. 

Reviewer: ```suggestion
and some extra arguments that will be formatted inside these replacement fields. 
```

Even if the format string is checked at compile-time, it is possible to have a mismatch between the format string and the arguments:

* The format string contains fewer replacement fields than the number of extra arguments:
  `std::format("{} {}", 1, 2, 3);`
* The format string uses indexes for the replacement fields, but one index is missing:
  `std::format("{0} {0} {2}", 1, 2, 3);`

> When that happens, the extra arguments are silently ignored. In the best scenario, it is dead code that could be removed, 
> but more probably, it is a typo and the output will not be the expected one.

Reviewer: ```suggestion
When that happens, the extra arguments are silently ignored. In the best scenario, it is dead code that could be removed, 
but more probably, it is a typo, and the output will not be the expected one.
```

> This rule raises an issue when the unused argument is the result of a complex expression (for instance a function call) or, 
> in the case of numbered replacement fields, when in addition of a missing index, another index appears several times.

Reviewer: ```suggestion
This rule raises an issue when the unused argument is the result of a complex expression (for instance a function call) or, 
in the case of numbered replacement fields, when in addition to a missing index, another index appears several times.
```
